evm/v0.5/contracts/dev/PushAggregator.sol

```diff
@@ -0,0 +1,195 @@
+pragma solidity 0.5.0;
+
+import "../Median.sol";
+import "../vendor/Ownable.sol";
+import "../interfaces/LinkTokenInterface.sol";
+
+/**
+ * @title The PushAggregator handles aggregating data pushed in from off-chain.
+ */
+contract PushAggregator is Ownable {
+
+  struct OracleStatus {
+    bool enabled;
+    uint256 lastReportedRound;
+  }
+
+  struct Round {
+    uint64 maxAnswers;
+    uint64 minAnswers;
+    uint128 paymentAmount;
+    int256[] answers;
+  }
+
+  int256 public currentAnswer;
+  uint256 public currentRound;
+  uint128 public paymentAmount;
+  uint128 public oracleCount;
+  uint64 public maxAnswerCount;
+  uint64 public minAnswerCount;
+
+  LinkTokenInterface private LINK;
+  mapping(address => OracleStatus) private oracles;
+  mapping(uint256 => Round) private rounds;
+
+  event NewRound(uint256 indexed number);
+  event AnswerUpdated(int256 indexed current, uint256 indexed round);
+
+  constructor(address _link, uint128 _paymentAmount)
+    public
+  {
+    LINK = LinkTokenInterface(_link);
+    setPaymentAmount(_paymentAmount);
+  }
+
+  function updateAnswer(uint256 _round, int256 _answer)
+    public
+    ensureValidRoundId(_round)
+    validateOracleRound(_round)
+  {
+    startNewRound(_round);
+    recordAnswer(_answer, _round);
+    updateRoundAnswer(_round);
+    deleteRound(_round);
+
+    require(LINK.transfer(msg.sender, paymentAmount), "LINK transfer failed");
```

Comment:
Just noting that you commented this needs to be changed to use the round's payment amount. We could also use a test that ensure the round's payment is used.